cmake/legate_helper_functions.cmake

```
@@ -0,0 +1,184 @@
+macro(legate_include_rapids)
+  if (NOT _LEGATE_HAS_RAPIDS)
+    if(NOT EXISTS ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+      file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-22.08/RAPIDS.cmake
```

Should probably update to (at least) `branch-22.10`, which added the ability for the user [to control](https://github.com/rapidsai/rapids-cmake/blob/branch-22.10/RAPIDS.cmake#L21-L24) the _actual_ version of `rapids-cmake` that is downloaded externally:

```shell
cmake -S . -B build -Drapids-cmake-version=23.04
```


```suggestion
      file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-23.02/RAPIDS.cmake
```

---

Also place the file in `LEGATE_RAPIDS.cmake` so that you don't get clobbered by any other dep / consumer that uses rapids and writes to `RAPIDS.cmake`

---

Good call, thank you